ffyaml/yaml.go

```diff
@@ -0,0 +1,22 @@
+package ffyaml
+
+import (
+	"fmt"
+	"io"
+
+	"gopkg.in/yaml.v2"
+)
+
+func Parser(r io.Reader, set func(name, value string) error) error {
+	var m map[string]string
+	d := yaml.NewDecoder(r)
+	if err := d.Decode(&m); err != nil && err.Error() != "EOF" {
+		return fmt.Errorf("error parsing YAML config: %v", err)
+	}
+	for key, val := range m {
+		if err := set(key, val); err != nil {
+			return err
+		}
+	}
```

Comment:
How will you support flags that can be set multiple times? Can you add support for something like YAML arrays or lists, similar to [this bit of code in the JSON parser](https://github.com/peterbourgon/ff/blob/52d851567994fcca73bf3668fc3099cfa7be3768/json.go#L22)?

Reply:
I know it's possible... but I don't really see the point. Doesn't fs only support singular values?

Reply:
Flags can be repeated, and if the value supports it, each instance of the flag can be saved separately. None of the default flag value types support that but it’s easy to write your own which does. I use this sort of thing frequently.

Reply:
See also https://godoc.org/github.com/carlmjohnson/flagext#Strings